src/patterns/ProgressWizard/components/InnerProgressWizard.tsx

```diff
@@ -0,0 +1,187 @@
+import { cloneElement, forwardRef, useEffect, useMemo, type ReactNode } from 'react';
+import ProgressIndicator from '../../../components/ProgressIndicator/ProgressIndicator';
+import { getCommonProps } from '../../../utils';
+import classNames from 'classnames';
+import Icon from '../../../components/Icon/Icon';
+import { useFormContext } from 'react-hook-form';
+import { Footer } from './ProgressWizardFooter';
+
+import { type PublicState, type FormStep, type ButtonLabels, type Handlers } from '../types';
+import { handleHiddenFields } from '../utils';
+import { useProgressWizardForm } from '../hooks/useProgressWizardForm';
+
+/**
+ * Props for the internal ProgressWizard layout component. Used to render the wizard UI and handle navigation.
+ * @property steps - Array of FormStep objects (wizard steps)
+ * @property currentStepIndex - Index of the current step (0-based)
+ * @property setCurrentStepIndex - Setter for current step index
+ * @property customHeader - Optional custom header ReactNode
+ * @property hideNavigation - If true, hides the default footer navigation (so you can implement your own)
+ * @property buttonLabels - Button labels for navigation (see type ButtonLabels)
+ * @property loadingState - Current loading state (see type LoadingState, aligns with a remix fetcher)
+ * @property setLoadingState - Setter for loading state
+ * @property action - Optional form action URL, irrelevant if `onFormSubmit` is supplied to the parent component.
+ * @property isFirstStep - True if this is the first step
+ * @property isLastStep - True if this is the last step
+ * @property className - Optional class for styling
+ * @private formId - ID for the form element, generated by the parent
+ *
+ * Inherits handler props from Handlers.
+ */
+
+export interface InnerProgressWizardProps extends Required<Handlers>, PublicState {
+  steps: FormStep[];
```

Comment:
instead of the `@property` docs, putting them above each prop in this interface usually results in better storybook documentation